includes/class-amp-theme-support.php

```diff
@@ -1745,6 +1745,30 @@ public static function ensure_required_markup( DOMDocument $dom, $script_handles
 			$amp_scripts[ $missing_script_handle ] = AMP_DOM_Utils::create_node( $dom, 'script', $attrs );
 		}
 
+		// Remove scripts that had already been added but couldn't be detected from output buffering.
+		foreach ( array_diff( array_keys( $amp_scripts ), $script_handles ) as $superfluous_script_handle ) {
+			$src = $amp_scripts[ $superfluous_script_handle ]->getAttribute( 'src' );
+
+			// Skip scripts unrelated to AMP.
+			if ( ! $src || 0 !== strpos( $src, 'https://cdn.ampproject.org/' ) ) {
```

Comment:
Nit: This could be made marginally faster via:

```suggestion
			if ( ! $src || 'https://cdn.ampproject.org/' !== substr( $src, 0, 27 ) ) {
```

(No need to search the _entire_ large string when we're just wanting to know if it `startsWith` another string.)

Reply:
In benchmarks with a 4800 character string, the end result is that these are pretty much exactly the same in terms of performance: http://www.spudsdesign.com/benchmark/index.php?t=strpos4&history

However, the `substr()` variant has a maintainability issue in that it requires you to make changes in two places if the string to check for changes: the string itself, and its character count need to be adapted separately - and you can even get it wrong by miscounting.

Therefore, I think the `strpos()` is the superior one.

Reply:
Yeah, it's fine. My wink was intended to acknowledge the exceedingly marginal performance impact.